test/api/line_items_test.rb

```diff
@@ -0,0 +1,160 @@
+require "minitest/autorun"
+require_relative "../test_helper"
+
+describe "line items API" do
+  describe "GET /api/clients/:client_id/line_items" do
+    before do
+      skip_if_impl_in %w(clojure_liberator crystal_kemal elixir_phoenix haskell_scotty kotlin_spark node_express node_hapi python_flask ruby_sinatra)
+
+      db.seed(<<-SQL)
+        INSERT INTO line_items SET
+          id=1,
+          client_id=1,
+          user_id=1,
+          start="2000-01-01T00:00:00Z",
+          finish="2000-01-01T02:30:00Z",
+          rate=100.00,
+          notes="Unassigned line item",
+          created_at="2000-01-01T00:00:00Z",
+          updated_at="2000-01-01T02:30:00Z",
+          invoice_id=NULL,
+          type="LineItem";
+      SQL
+
+      @token = sign_up_and_authorize_account({
+        name: "Micah Geisel",
+        email: "micah@botandrose.com",
+        password: "omgponies",
+      })
+    end
+
+    it "lists line items for the given client" do
+      response = get("/api/clients/1/line_items", "Authorization" => "Bearer #{@token}")
```

Comment:
The JSON API recommendation makes me want to go with dashes, but I think its a can of worms I'd rather leave closed. If we kebabed "line-items", I think we'd have to go through and kebab all the (many) attribute names, which would break all the impls, so I'm inclined to stick with underscores. Next time I start a JSON API project, I'll start with kebab case from the get-go.